src/cosmicqc/scdataframe.py

```diff
@@ -0,0 +1,164 @@
+        __getattr__():
+            Returns the underlying attributes of the pandas DataFrame.
+        __getitem__():
+            Returns slice of data from pandas DataFrame.
+    """
+
+    def __init__(
+        self: SCDataFrame_type,
+        data: Union[SCDataFrame_type, pd.DataFrame, str, pathlib.Path],
+        **kwargs: Dict[str, Any],
+    ) -> None:
+        """
+        Initializes the SCDataFrame with either a DataFrame or a file path.
+
+        Args:
+            data (Union[pd.DataFrame, str]):
+                The data source, either a pandas DataFrame or a file path.
+            **kwargs:
+                Additional keyword arguments to pass to the pandas read functions.
+        """
+
+        if isinstance(data, SCDataFrame):
+            # if data is an instance of SCDataFrame, use its data_source and data
+            self.data_source = data.data_source
+            self.data = data.data
+
+        elif isinstance(data, pd.Series):
+            # if data is a pd.Series, remember this within the data_source attr
+            self.data_source = "pandas.Series"
+            # also cast the series to a dataframe
+            self.data = pd.DataFrame(data)
+
+        elif isinstance(data, pd.DataFrame):
+            # if data is a pd.DataFrame, remember this within the data_source attr
+            self.data_source = "pandas.DataFrame"
+            self.data = data
+
+        elif isinstance(data, (pathlib.Path, str)):
+            # if the data is a string, remember the original source
+            # through a data_source attr
+            self.data_source = data
+
+            # interpret the data through pathlib
+            data_path = pathlib.Path(data)
+
+            # Read the data from the file based on its extension
+            if (
+                data_path.suffix == ".csv"
+                or data_path.suffix in (".tsv", ".txt")
+                or data_path.suffixes == [".csv", ".gz"]
+            ):
+                # read as a CSV, CSV.GZ, .TSV, or .TXT file
+                self.data = pd.read_csv(data, **kwargs)
+            elif data_path.suffix == ".parquet":
+                # read as a Parquet file
+                self.data = pd.read_parquet(data, **kwargs)
+            else:
+                raise ValueError("Unsupported file format for SCDataFrame.")
+        else:
+            raise ValueError("Unsupported data type for SCDataFrame.")
+
+    def export(
+        self: SCDataFrame_type, file_path: str, **kwargs: Dict[str, Any]
+    ) -> None:
+        """
+        Exports the underlying pandas DataFrame to a file.
+
+        Args:
+            file_path (str): The path where the DataFrame should be saved.
+            **kwargs: Additional keyword arguments to pass to the pandas to_* methods.
+        """
+
+        data_path = pathlib.Path(file_path)
+
+        # export to csv
+        if ".csv" in data_path.suffixes:
+            self.data.to_csv(file_path, **kwargs)
```

Comment:
Should we include the ability to export also as a compressed file as well?

Reply:
Thanks for checking on this, we should be able to export compressed files as-is based on inference or `kwarg` usage, for example using [`pandas.DataFrame.to_csv(...,compression='infer',...)`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html).